### ISSUE TITLE
Resizing top/left edge of some apps makes them snap back when moved

### ISSUE THREAD

**Issue by git-f0x** [pop-os/cosmic-comp#304]
If the top/left edges of a floating window are the last resized using the keyboard (`Super + R + arrows` / `Super + Shift + R + arrows`) before resizing is stopped, the window becomes anchored to the resized **Y** or **X** coordinate, respectively.

If the top edge was resized, dragging the window down/up from its position and then releasing the mouse button makes the window return to its original **Y** coordinate (keeping its new **X** coordinate). This persists when the window is moved to a screen edge with `Super + Shift + arrows` or maximized, and makes the window go partially off-screen.
The left edge behaves similarly, just that it returns to its original **X** and keeps the new **Y** coordinate.

Resizing with a mouse also reproduces the issue, but only once.
If resizing with the keyboard, it is necessary to resize the bottom or right edges to resolve the issue (or any edge with a mouse, but the top and left will cause the window to snap back once).

I'm assuming **X** is the axis of the longer screen edge (left/right), and **Y** the shorter (up/down), on a landscape screen.

Also, something that might be related: when a window's top/left edge touches a screen edge while the keyboard resize keys are being held (continuously), the window jitters, which doesn't happen with the bottom/right edges.

Edit: Upon further testing, resizing with the keyboard doesn't make Xwayland apps anchored (i.e. only affects apps running natively with Wayland), but resizing with the mouse still makes them snap back once, and they also jitter against screen edges.

Recording of the issue:

https://github.com/pop-os/cosmic-comp/assets/150025636/0d178cfb-4eaa-4dcd-80a7-025d733e3a28

**Comment by git-f0x** [comment]
Upon some more testing, at least with the latest commits, I can't reliably reproduce the mouse portion of this issue on anything other than Firefox running on Wayland (Firefox on Xwayland, and Xwayland apps in general can't reliably reproduce this issue now, other than the jitter against the top/left edges), at least of the libcosmic, GTK and QT apps I tested.
Though admittedly, I'm not currently using my usual system.
Resizing with the keyboard still affects all Wayland apps.

**Comment by Drakulix** [comment]
Should be fixed by https://github.com/pop-os/cosmic-comp/commit/01c69cb7eae006d305a3aaf2c43d700c206dc0b7

**Comment by git-f0x** [comment]
Everything seems to be fixed other than the jitter against top/left edges and Wayland Firefox snaps back once after resize with keyboard/mouse. No idea why specifically Firefox, might be an interaction with this specific hardware.

**Comment by git-f0x** [comment]
Firefox and Nautilus snap back once when the top/left edges are resized on my main system as well.

**Comment by git-f0x** [comment]
It seems Qt apps (Dolphin, KWrite) also snap back once when the top (and occasionally left) edge is resized.

**Comment by git-f0x** [comment]
Don't know if this should be reopened or a separate issue created, since it still snaps back once with some apps @Drakulix.

**Comment by git-f0x** [comment]
It seems the apps no longer snap back, but it still has similar effects if the window is maximized after being resized in those directions.

**Comment by WatchMkr** [comment]
I'm going to close this one. Testing did expose another issue. The resize arrows are below the window in floating mode.

**Comment by git-f0x** [comment]
It's not fixed yet, at least with Firefox @WatchMkr.
E.g. resize a the top edge of a floating Firefox window, then maximize it.
